aviary/subsystems/propulsion/turboprop_model.py

```diff
@@ -0,0 +1,193 @@
+        # accept target scaling variable as an option, skipping for now
+        if type(shp_model) is not EngineDeck:
+            shp_model_pre_mission = shp_model.build_pre_mission(aviary_inputs, **kwargs)
+            turboprop_group.add_subsystem(shp_model_pre_mission.name,
+                                          subsys=shp_model_pre_mission,
+                                          promotes=['*'])
+
+        if self.propeller_model is not None:
+            prop_model_pre_mission = prop_model.build_pre_mission(
+                aviary_inputs, **kwargs)
+            turboprop_group.add_subsystem(prop_model_pre_mission.name,
+                                          subsys=prop_model_pre_mission,
+                                          promotes=['*'])
+
+        return turboprop_group
+
+    def build_mission(self, num_nodes, aviary_inputs, **kwargs):
+        shp_model = self.shaft_power_model
+        prop_model = self.propeller_model
+        turboprop_group = om.Group()
+
+        shp_model_mission = shp_model.build_mission(num_nodes, aviary_inputs, **kwargs)
+        turboprop_group.add_subsystem(shp_model.name,
+                                      subsys=shp_model_mission,
+                                      promotes_inputs=['*'],
+                                      promotes_outputs=['*', (Dynamic.Mission.THRUST, 'turboshaft_thrust')])
+
+        # ensure uncorrected shaft horsepower is avaliable
+        # TODO also make sure corrected is avaliable
+        # TODO see if this can be done for non-EngineDecks
+        if type(shp_model_mission) is EngineDeck:
+            if EngineModelVariables.SHAFT_POWER not in shp_model_mission.engine_variables:
+                turboprop_group.add_subsystem('uncorrect_shaft_power',
+                                              subsys=UncorrectShaftPower(num_nodes=num_nodes,
+                                                                         aviary_options=self.options),
+                                              promotes_inputs=[Dynamic.Mission.SHAFT_POWER_CORRECTED,
+                                                               Dynamic.Mission.TEMPERATURE,
+                                                               Dynamic.Mission.STATIC_PRESSURE,
+                                                               Dynamic.Mission.MACH],
+                                              promotes_outputs=[Dynamic.Mission.SHAFT_POWER]),
+
+        if prop_model is not None:  # must assume user-provide propeller group has everything it needs
+            prop_model_mission = prop_model.build_mission(
+                num_nodes, self.options, **kwargs)
+            turboprop_group.add_subsystem(prop_model.name,
+                                          subsys=prop_model_mission,
+                                          promotes_inputs=['*'],
+                                          promotes_outputs=['*'])
```

Comment:
Can we assume that user provided models have a build_mission function? Seems like it might be better to expect them to just pass a preconfigured subsystem